src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -127,15 +127,21 @@ bool Series::saveToFolder(const std::string& folder) const
 
     bool ret = true;
 
-    auto checkWrite = [&ret, &folder](const std::string& name, const std::vector<double>& content) {
-        ret = writeVectorToFile(folder + SEP + name, content) && ret;
+    auto checkWrite = [&ret, &folder]
```

Comment:
About **checkWrite** : 
This function does 2 things : it checks and it saves to folder.
Wouldn't it be more clear if we had : 
```C++
if(! hasOnlyDefaultValues(maxInjectionModulation, 1.0))
    ret = saveTimeSeriesToFile(maxInjectionModulation, "PMAX-injection.txt") && ret;

// Here : same for other time series 
``` 
?
Note that each functions has less arguments (=2), instead of 5 for **checkWrite** (if we count lambda capture)

Reply:
remove the lambda means having a one line function with those args and needing:
`saveTimeSeriesToFile(const std::string& folder, const std::string& filename, const std::vector<double>& timeSeries)`
and will need to handle ret on the other function like this:
`ret = saveTimeSeriesToFile() && ret`
This was why we used a lambda in the first place

Reply:
Or we can duplicate code and have 5*2 lines doing the same thing with different var

Reply:
I guess you're concerned about duplication and concision.
I'm more concerned about clarity, short function making one thing with small number of arguments.
**checkWrite** makes 2 things (and it's in the name **checkAndWrite**), it's usually considered as a code smell.